src/gen_caml.ml

```diff
@@ -268,10 +276,32 @@ let make_sql l =
   Buffer.add_string b ")";
   Buffer.contents b
 
+let in_var_module label = function
+  | Sql.Type.Tuple _ | Any | Unit _ as typ -> failwith @@ sprintf "invalid IN var type %s in @%s" (Sql.Type.show typ) label
+  | typ -> Sql.Type.to_string typ
+
+let gen_in_substitution var =
+  if Option.is_none var.id.label then failwith "empty label in IN param";
+  match var.typ with
+  | Sql.Type.Unit _ | Int | Text | Blob | Float | Bool | Datetime | Decimal | Any | Tuple (Any | Tuple _) as typ ->
+    failwith @@ sprintf "invalid inferred IN var type %s (%s)" (Sql.Type.show typ) (Option.get var.id.label)
```

Comment:
this cannot possibly happen, by construction, ie it is more of assert than error for the user? (nothing to change, just asking to understand)

Reply:
Yes, basically "in variables" must have `Tuple x` type (where x is not a Tuple itself), so this is just an assert with a more informative message for the `sqlgg` developer.
 
This could be removed by making `param` polymorphic and switching to polymorphic variants in the type language, but it requires extra annotations (or patience) to type.

Another idea is adding a "is IN var" flag to `param`. That might allow to reuse `F` as is.